crates/swc_ecma_lints/src/rules/no_prototype_builtins.rs

```diff
@@ -0,0 +1,151 @@
+            }
+            LintRuleReaction::Warning => {
+                handler.struct_span_warn(span, &message).emit();
+            }
+            _ => {}
+        });
+    }
+
+    fn extend_chain(&mut self, span: Span, atom: Atom) {
+        if self.call_info.method_span.is_none() {
+            self.call_info.method_span = Some(span);
+        }
+
+        self.call_info.chain.push(atom);
+    }
+
+    fn extract_path(&mut self, expr: &Expr) {
+        match expr {
+            Expr::Member(member) => {
+                match &member.prop {
+                    MemberProp::Ident(ident) => {
+                        self.extend_chain(ident.span, ident.sym.clone());
+                    }
+                    MemberProp::Computed(computed_prop_name) => {
+                        match computed_prop_name.expr.as_ref() {
+                            Expr::Lit(_) | Expr::Tpl(_) | Expr::Paren(_) | Expr::Seq(_) => {
+                                self.extract_path(&computed_prop_name.expr);
+                            }
+                            _ => {}
+                        }
+                    }
+                    _ => {}
+                }
+
+                self.extract_path(member.obj.as_ref());
+            }
+            Expr::OptChain(OptChainExpr { base, .. }) => {
+                if let Some(member_expr) = base.as_member() {
+                    if let Some(ident) = member_expr.prop.as_ident() {
+                        self.extend_chain(ident.span, ident.sym.clone());
+                    }
+
+                    self.extract_path(member_expr.obj.as_ref());
+                }
+            }
+            Expr::Paren(ParenExpr { expr, .. }) => {
+                self.extract_path(expr.as_ref());
+            }
+            Expr::Seq(SeqExpr { exprs, .. }) => {
+                self.extract_path(exprs.last().unwrap().as_ref());
+            }
+            Expr::Lit(Lit::Str(lit_str)) => {
+                self.extend_chain(lit_str.span, lit_str.value.clone());
+            }
+            Expr::Tpl(tpl) => {
+                if tpl.exprs.is_empty() && tpl.quasis.len() == 1 {
+                    self.extend_chain(tpl.span, tpl.quasis[0].raw.clone());
+                }
+            }
+            Expr::Ident(ident) => {
+                self.extend_chain(ident.span, ident.sym.clone());
+            }
+            _ => {}
+        }
+    }
+
+    fn check(&mut self, expr: &Expr) {
+        let prev_call_info = std::mem::take(&mut self.call_info);
+
+        self.extract_path(expr);
+
+        if self.call_info.chain.len() > 1 {
+            let method_name = self.call_info.chain[0].as_str();
+
+            if METHODS.contains(&method_name) {
+                self.emit_error(self.call_info.method_span.unwrap(), method_name);
+            }
+        }
+
+        self.call_info = prev_call_info;
+    }
+}
+
+impl Visit for NoPrototypeBuiltins {
+    fn visit_opt_chain_base(&mut self, opt_chai_base: &OptChainBase) {
```

Comment:
This looks like a type

Reply:
I've checked without this visitor and part of my test are failed.

I think here is bug, because `visit_opt_chain_base` actually never visited, but without defining `visit_opt_chain_base` visitor `visit_call_expr` skip reaction on case like `foo?.hasOwnProperty('bar')`

Reply:
I mean `opt_chai_base`

Reply:
Oh... I did reading u comment in a little hurry and read "type" instead of "typo" sorry

Reply:
🤣 I wrote `type`, sorry. It was also a typo.

Reply:
got morning damage 🤣

Reply:
Anyway it was useful. I'd fixed code after broken merge and added noop_visit_type to prevent extra job